lib/matplotlib/tests/test_backends_interactive.py

```diff
@@ -623,6 +623,23 @@ def test_blitting_events(env):
     assert 0 < ndraws < 5
 
 
+def _fallback_check():
+    import IPython.core.interactiveshell as ipsh
+    import matplotlib.pyplot
+    ipsh.InteractiveShell.instance()
+    matplotlib.pyplot.figure()
+
+
+def test_fallback_to_different_backend():
+    pytest.importorskip("IPython")
+    import subprocess
+    # Runs the process that caused the GH issue 23770
+    # making sure that this doesn't crash
+    # since we're supposed to be switching to a different backend instead.
+    response = _run_helper(_fallback_check, timeout=_test_timeout)
+    assert response != subprocess.CalledProcessError
```

Comment:
The run helper passes `check=True`, which causes `CalledProcessError` to be raised; I don't understand why this is checking the return value, which should be a `subprocess.CompletedProcess`.

Reply:
You're right. I just thought we had to check for it. Should I leave the test at this, then? It seems to do the same thing (fail without the fix, pass with the fix).
``` python3
def test_fallback_to_different_backend():
    pytest.importorskip("IPython") 
    _run_helper(_fallback_check, timeout=_test_timeout)
```

Reply:
All exceptions are errors; the `assert` can never be False, so there's no need for it.

Reply:
Ok then, thank you, I'll leave it without the assert!